src/callbacks.jl

```diff
@@ -560,7 +560,37 @@ end
   end
   event_idx = 1
 
-  event_occurred,interp_index,Θs,prev_sign,prev_sign_index,event_idx
+  event_occurred,interp_index,ts,prev_sign,prev_sign_index,event_idx
+end
+
+# rough implementation, needs multiple type handling
+# always ensures that if r = bisection(f, (x0, x1))
+# then either f(nextfloat(r)) == 0 or f(nextfloat(r)) * f(r) < 0
+function bisection(f, tup, tdir)
+  x0, x1 = tup
+  @assert f(x0) * f(x1) < 0.0
+  left = x0
+  right = x1
+  prevfloat_tdir = (tdir === 1.0 ? prevfloat : nextfloat)
```

Comment:
What's the reason for not writing
```suggestion
  prevfloat_tdir = isone(tdir) ? prevfloat : nextfloat
```

Reply:
Oh yes, AD test would fail otherwise, my bad

Reply:
The `@assert` statements could be problematic as well, I guess, but I assumed these are mainly for debugging right now?

Reply:
I can change them to proper errors. But those are very important assertions, in case of any edge case, we need a solid starting point to debug